### PR TITLE
Invitation modal

### PR DESCRIPTION
**Pull request by Ducica**

### CODE REVIEW

**Review by mirekys — CHANGES_REQUESTED** [review]
Screenshot 2024-09-17 at 12 17 57

**Comment by mirekys** [comment]
Screenshot 2024-09-17 at 12 18 21

**Review by mirekys — CHANGES_REQUESTED** [review]
Missing padding on `People` tab
Screenshot 2024-09-17 at 12 18 52

**Comment by Ducica** [comment]
> Screenshot 2024-09-17 at 12 18 21

This is hack for how they changed the rich editor implementation in RDM too, it is very ugly, it was in the merged PR in oarepo ui too. https://github.com/oarepo/oarepo-ui/commit/38c7dad3ed3cefa7681e2cefecdf160f27b8b75d

**Review by mirekys — CHANGES_REQUESTED** [review]
Insufficient input validation, allows for invalid input:
Screenshot 2024-09-17 at 13 24 55
Screenshot 2024-09-17 at 13 26 09

Screenshot 2024-09-17 at 13 26 34

File: oarepo_communities/ui/communities_components/semantic-ui/js/communities_components/components/CommunityInvitationsModal/CommunityInvitationsModal.jsx [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+                <Form.Field>
+                  <TextAreaField
+                    fieldPath="membersEmails"
+                    required
+                    autoFocus
+                    label={
+                      <FieldLabel
+                        label={i18next.t("Members")}
+                        icon={
+                          validEmailsCount + invalidEmailsCount > 0 &&
```
Comment (mirekys) [inline oarepo_communities/ui/communities_components/semantic-ui/js/communities_components/components/CommunityInvitationsModal/CommunityInvitationsModal.jsx:128]: valid > 0 && invalid === 0 would be more understandable :)

File: oarepo_communities/ui/communities_components/semantic-ui/js/communities_components/components/CommunityInvitationsModal/CommunityInvitationsModal.jsx [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+                        <List.Content>
+                          <List.Header>{u.title}</List.Header>
+                          <List.Description>{u.description}</List.Description>
+                        </List.Content>
+                      </List.Item>
+                    ))}
+                  </List>
+                  <ErrorLabel fieldPath="role" />
+                </Form.Field>
+                <RichInputField
```
Comment (mirekys) [inline oarepo_communities/ui/communities_components/semantic-ui/js/communities_components/components/CommunityInvitationsModal/CommunityInvitationsModal.jsx:179]: Can we use oarepo-ui rich editor component?

File: oarepo_communities/ui/communities_components/semantic-ui/js/communities_components/components/CommunityInvitationsModal/CommunityInvitationsModal.jsx [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+                  initialValue=""
+                />
+              </Form>
+              {successMessage && <Message positive>{successMessage}</Message>}
+              {httpError && <Message negative>{httpError}</Message>}
+            </Modal.Content>
+            <Modal.Actions>
+              <Button
+                onClick={() => {
+                  close();
```
Comment (mirekys) [inline oarepo_communities/ui/communities_components/semantic-ui/js/communities_components/components/CommunityInvitationsModal/CommunityInvitationsModal.jsx:194]: On close we should also reset complete modal state (such as messages shown)

File: oarepo_communities/ui/communities_components/semantic-ui/js/communities_components/components/CommunityInvitationsModal/CommunityInvitationsModal.jsx [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+                  resetForm();
+                }}
+                floated="left"
+              >
+                <Icon name="remove" /> {i18next.t("Cancel")}
```
Comment (mirekys) [inline oarepo_communities/ui/communities_components/semantic-ui/js/communities_components/components/CommunityInvitationsModal/CommunityInvitationsModal.jsx:199]: label should say `Close` (we just close the dialog, not cancel any request in progress)

**Review by mirekys — CHANGES_REQUESTED** [review]

File: oarepo_communities/ui/communities_components/semantic-ui/js/communities_components/components/CommunityInvitationsModal/CommunityInvitationsModal.jsx (hunk 2 above)
Comment (Ducica) [reply oarepo_communities/ui/communities_components/semantic-ui/js/communities_components/components/CommunityInvitationsModal/CommunityInvitationsModal.jsx:186]: @mirekys It is not possible unfortunately, due to the fact we don't have formConfig there.

File: oarepo_communities/ui/communities_components/semantic-ui/js/communities_components/components/CommunityInvitationsModal/CommunityInvitationsModal.jsx (hunk 1 above)
Comment (Ducica) [reply oarepo_communities/ui/communities_components/semantic-ui/js/communities_components/components/CommunityInvitationsModal/CommunityInvitationsModal.jsx:135]: @mirekys agreed :)

**Comment by Ducica** [comment]
> Insufficient input validation, allows for invalid input: Screenshot 2024-09-17 at 13 24 55 Screenshot 2024-09-17 at 13 26 09
> 
> Screenshot 2024-09-17 at 13 26 34
> Validation needs to be much stricter - split into lines, enforce that each line matches just the exact regex pattern of email or display name <email>

I "borrowed" the code from that library. It might seem a bit overcomplicated, but apart from the regexp itself, it does make sense and it seems to work quite nicely. 

ISFQDN checks if the mail has fully qualified domain name (part after @), the byte length is just a limitation for the length of mail and display name and the main function utilizes these functions and the regexp to finally validate the mail.

**Comment by Ducica** [comment]
@mirekys There are some warnings from sonarlint, about the use of control characters in regexps, though in the validator.js https://github.com/validatorjs/validator.js/blob/ff56dcf5ad16abc4127528eafae559ac716863fb/src/lib/isEmail.js#L64, this warning is also disabled in eslint, so I assume that it is OK to use. If you can take a look as well please I would be grateful. Thanks

**Comment by mirekys** [comment]
Screenshot 2024-10-11 at 10 21 47

**Comment by mirekys** [comment]
Screenshot 2024-10-11 at 10 26 30

**Comment by Ducica** [comment]
@mirekys Hey, thanks a lot for feedback. I pushed the changes.

**Review by mirekys — APPROVED** [review]

**Pull request merged**